lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystemE2E.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure.Authentication;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Xunit;
+using System.Threading;
+using Org.Apache.REEF.IO.FileSystem.AzureDataLake;
+using Microsoft.Azure.DataLake.Store;
+using Org.Apache.REEF.Tang.Interface;
+using System.IO;
+using System.Linq;
+
+namespace Org.Apache.REEF.IO.Tests
+{
+    /// <summary>
+    /// E2E tests for AzureDataLakeFileSystem.
+    /// These tests require the person running the test to fill in credentials.
+    /// </summary>
+    public sealed class TestAzureDataLakeFileSystemE2E : IDisposable
+    {
+        private const string SkipMessage = "Fill in credentials before running test"; // Use null to run tests
+        private const string ContentsText = "hello";
+        private readonly IFileSystem _fileSystem;
+        private readonly AdlsClient _adlsClient;
+        private readonly string _defaultFolderName;
+
+        public TestAzureDataLakeFileSystemE2E()
+        {
+            // Service principal / appplication authentication with client secret / key
+            // Use the client ID of an existing AAD "Web App" application.
+            // Fill in before running test!
+            const string AdlsAccountName = "#####.azuredatalakestore.net";
+            const string Tenant = "microsoft.onmicrosoft.com";
+            const string TokenAudience = @"https://datalake.azure.net/";
+            const string ClientId = "c2897d56-5eef-4030-8b7a-46b5c0acd05c";
+            const string SecretKey = "##########";
```

Comment:
Similar to ClientId, give an example so that the user would know what it looks like.